query/workmanager.go

```diff
@@ -0,0 +1,464 @@
+package query
+
+import (
+	"container/heap"
+	"errors"
+	"sync"
+	"time"
+)
+
+const (
+	// minQueryTimeout is the timeout a query will be initially given. If
+	// the peer given the query fails to respond within the timeout, it
+	// will be given to the next peer with an increased timeout.
+	minQueryTimeout = 2 * time.Second
+
+	// maxQueryTimeout is the maximum timeout given to a single query.
+	maxQueryTimeout = 32 * time.Second
+)
+
+var (
+	// ErrWorkManagerShuttingDown will be returned in case the WorkManager
+	// is in the process of exiting.
+	ErrWorkManagerShuttingDown = errors.New("WorkManager shutting down")
+)
+
+type batch struct {
+	requests []*Request
+	options  *queryOptions
+	errChan  chan error
+}
+
+// Worker is the interface that must be satisfied by workers managed by the
+// WorkManager.
+type Worker interface {
+	// Run starts the worker and associates it with the given peer. peer is
```

Comment:
This comment is now outdated.

Reply:
Fixed: https://github.com/lightninglabs/neutrino/pull/179/commits/36c12ad48585b21ba8e1c6fa73b9c6c29edf0aaa